### PR TITLE
[3701] - send course vacancies partially updated notification

### PR DESCRIPTION
**Pull request by TobyRet**
### Context
Given I am a user associated with an accredited body
I want to be notified when a provider whom I accredit updates the vacancy status of some locations
So that I can (update my records? and) do amazing things

### Changes proposed in this pull request
- If a course has multiple locations and some of these locations has their vacancy statuses updated, this will
trigger a new 'course vacancies partially updated' notification
- The `send_vacancies_updated_notification` endpoint has been refactored to take an array of vacancy statuses. This means the API can more easily determine which notification to send
- Related Publish PR - https://github.com/DFE-Digital/publish-teacher-training/pull/1289

### Guidance to review
- This PR introduced a new notification - horribly titled `course_vacancies_partially_updated_email` - there maybe a better name for this. Please improve on this if you can.
- To test this out manually (locally) you will need to run against the [Publish PR](https://github.com/DFE-Digital/publish-teacher-training/pull/1289). You will also need to ensure that the user you sign in as is associated with the appropriate accrediting body and has notifications turned on. _Hint_ : Try testing with accrediting body  `L51` (Liverpool John Moored University) and accredited provider `2KA` (Archbishop Blanch School). To trigger the notification go to a course that is accredited by the accrediting body and edit vacancies. This will need to be a course that has multiple locations. Edit the vacancies for this course but only turn off / on vacancies from some of the locations. 

### View a notification
Example of `course vacancies partially updated notification` can be viewed at https://www.notifications.service.gov.uk/services/022acc23-c40a-4077-bbd6-fc98b2155534/notification/0b4c932a-f594-45f1-ba3f-fe259c54383f

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[[S]Vacancies - some or all open/closed (multiple locations)](https://trello.com/c/PHamO3fF/3701-svacancies-some-or-all-open-closed-multiple-locations)

File: app/mailers/course_vacancies_updated_email_mailer.rb [hunk 1]
```diff
@@ -17,6 +17,22 @@ def course_vacancies_updated_email(course:, user:, datetime:, vacancies_filled:)
     mail(to: user.email)
   end
 
+  def course_vacancies_partially_updated_email(course:, user:, datetime:, vacancies_opened:, vacancies_closed:)
```
Comment (TobyRet) [inline app/mailers/course_vacancies_updated_email_mailer.rb:20]: Is there a better name for `course_vacancies_partially_updated_email` 🤔

File: app/mailers/course_vacancies_updated_email_mailer.rb [hunk 2]
```diff
@@ -17,6 +17,22 @@ def course_vacancies_updated_email(course:, user:, datetime:, vacancies_filled:)
...
+    set_template(Settings.govuk_notify.course_vacancies_partially_updated_email_template_id)
```
Comment (TobyRet) [inline app/mailers/course_vacancies_updated_email_mailer.rb:21]: https://github.com/DFE-Digital/teacher-training-api/pull/1500/files#r470492291

File: app/mailers/course_vacancies_updated_email_mailer.rb (hunk 1 above)
Comment (darokel) [reply app/mailers/course_vacancies_updated_email_mailer.rb:20]: A few options:

1) Just shorten and remove the updated/email word. We already have this in the class name `CourseVacanciesUpdatedEmailMailer`, so you can build on this with a method name like `partially_updated`.
2) Building on from the first point, use namespacing for more descriptive/simple email classes which deal with one particular type of email ie `NotificationEmails::CourseVacancies::PartiallyUpdated`

Just some options :)

File: spec/mailers/course_vacancies_updated_email_mailer_spec.rb [hunk 3]
```diff
@@ -4,71 +4,130 @@
...
     course
     mail
   end
 
-  context "sending an email to a user" do
-    it "sends an email with the correct template" do
-      expect(mail.govuk_notify_template).to eq(Settings.govuk_notify.course_vacancies_updated_email_template_id)
-    end
+  context "course vacancies updated email" do
+    let(:mail) {
```
Comment (darokel) [inline spec/mailers/course_vacancies_updated_email_mailer_spec.rb:14]: can we use `do end` for multiline blocks

File: app/mailers/course_vacancies_updated_email_mailer.rb (hunk 1 above)
Comment (TobyRet) [reply app/mailers/course_vacancies_updated_email_mailer.rb:20]: Like those suggestions @darokel - will make the changes

**Pull request merged**